src/components/20-molecules/file-upload/index.js

```diff
@@ -205,27 +241,46 @@ class AXAFileUpload extends LitElement {
     this.requestUpdate();
   }
 
+  removeValidOrFaultyFileFromArray(index, files, isValid) {
+    const clonedFile = [...files];
+    if (isValid) {
+      clonedFile.splice(index, 1);
+    } else {
+      clonedFile.splice(index - this.validOriginalFiles.length, 1);
```

Comment:
Calculate index before calling that function. So you do not need an extra parameter "isValid" and just use parameters at your function that are described at the header in your brackets.

Reply:
done